compiler/damlc/daml-stdlib-src/DA/Internal/Desugar.daml

```diff
@@ -45,8 +48,16 @@ class HasInterfaceTypeRep i where
 class HasInterfaceTypeRep i => Implements t i where
   toInterface : t -> i
   fromInterface : i -> Optional t
-  toInterfaceContractId : ContractId t -> ContractId i
-  fromInterfaceContractId : ContractId i -> Update (Optional (ContractId t))
+
+toInterfaceContractId : forall i t. Implements t i => ContractId t -> ContractId i
```

Comment:
Note that this commit also changes the order of `toInterfaceContractId`'s type arguments, so we drop a few `@_`s in tests and generated code